python-avd/tests/pyavd/cv/client/test_async_decorators.py

```diff
@@ -96,14 +141,371 @@ async def test_msg_size_handler_invalid_fuction_return_type() -> None:
+        "logs_qty",
+        "log_patterns",
+        "inner_exception",
+        "outer_exception",
+        "outer_exception_patterns",
+    ),
+    [
+        pytest.param(
+            1,
+            1,
+            1,
+            [
+                "Attempt 1/6 to execute call '.*' returned '.*'\\. Retrying in [0-9]+s",
+            ],
+            GRPCError(Status.UNAVAILABLE),
+            does_not_raise(),
+            [],
+            id="ONE_GRPC_STATUS_UNAVAILABLE_FAILURE",
+        ),
+        pytest.param(
+            3,
+            3,
+            3,
+            [
+                "Attempt 1/6 to execute call '.*' returned '.*'\\. Retrying in [0-9]+s.*",
+                "Attempt 2/6 to execute call '.*' returned '.*'\\. Retrying in [0-9]+s.*",
+                "Attempt 3/6 to execute call '.*' returned '.*'\\. Retrying in [0-9]+s.*",
+            ],
+            GRPCError(Status.UNAVAILABLE),
+            does_not_raise(),
+            [],
+            id="THREE_GRPC_STATUS_UNAVAILABLE_FAILURES",
+        ),
+        pytest.param(
+            6,
+            5,
+            6,
+            [
+                "Attempt 1/6 to execute call '.*' returned '.*'\\. Retrying in [0-9]+s.*",
+                "Attempt 2/6 to execute call '.*' returned '.*'\\. Retrying in [0-9]+s.*",
+                "Attempt 3/6 to execute call '.*' returned '.*'\\. Retrying in [0-9]+s.*",
+                "Attempt 4/6 to execute call '.*' returned '.*'\\. Retrying in [0-9]+s.*",
+                "Attempt 5/6 to execute call '.*' returned '.*'\\. Retrying in [0-9]+s.*",
+                "Attempt 6/6 to execute call '.*' failed.",
+            ],
+            GRPCError(Status.UNAVAILABLE),
+            pytest.raises(CVGRPCStatusUnavailable),
+            ["Status\\.UNAVAILABLE: 14"],
+            id="SIX_GRPC_STATUS_UNAVAILABLE_FAILURES",
+        ),
+        pytest.param(
+            7,
+            5,
+            6,
+            [
+                "Attempt 1/6 to execute call '.*' returned '.*'\\. Retrying in [0-9]+s.*",
+                "Attempt 2/6 to execute call '.*' returned '.*'\\. Retrying in [0-9]+s.*",
+                "Attempt 3/6 to execute call '.*' returned '.*'\\. Retrying in [0-9]+s.*",
+                "Attempt 4/6 to execute call '.*' returned '.*'\\. Retrying in [0-9]+s.*",
+                "Attempt 5/6 to execute call '.*' returned '.*'\\. Retrying in [0-9]+s.*",
+                "Attempt 6/6 to execute call '.*' failed.",
+            ],
+            GRPCError(Status.UNAVAILABLE),
+            pytest.raises(CVGRPCStatusUnavailable),
+            ["Status\\.UNAVAILABLE: 14"],
+            id="SEVEN_GRPC_STATUS_UNAVAILABLE_FAILURES",
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    ("grpc_method", "extra_args"),
+    [
+        pytest.param("msgsize_unlimited_grpc_method", {}, id="UNLIMITED_SIZE_GRPC_METHOD"),
+        pytest.param("msgsize_limited_grpc_method", {"field": [0]}, id="LIMITED_SIZE_GRPC_METHOD"),
+    ],
```

Comment:
IMO this is too "clever". Can we simplify this by just calling both in the test and have different asserts for each. Alternatively you can write different tests for each and then set the list of params as a constant that can be reused.